sklearn/metrics/tests/test_common.py

```diff
@@ -653,8 +670,12 @@ def check_single_sample(name):
     metric = ALL_METRICS[name]
 
     # assert that no exception is thrown
-    for i, j in product([0, 1], repeat=2):
-        metric([i], [j])
+    if name in METRICS_WITH_NON_ZERO_Y:
+        for i, j in product([1, 2], repeat=2):
```

Comment:
You might want to note why other metrics fail with [1,2]

Reply:
@jnothman, do you suggest that I add a comment explaining the reason of the failure? or explain that in the description of the pull request above?

Reply:
A comment, so that this code is explicable to the reader

Reply:
@jnothman, while trying to reproduce the error to prepare the comment, I discovered that it's I mistakingly thought that the errors originated from this test case. 
 
 In turned out that using [1, 2] works for all tests cases. So, I am changing the code accordingly.